### ISSUE TITLE
Make ConcurrentTasks class log errors when joined.

### ISSUE THREAD

**Issue by artem-shelkovnikov** [elastic/connectors-python#789]
### Problem Description
ConcurrentTasks class is used to run multiple tasks in parallel, but if a task raises an error no piece of code is logging these errors.

### Proposed Solution
Make the code in ConcurrentTasks that joins tasks check the results and log errors for all tasks that finished with an exception

### Alternatives
Make the caller of ConcurrentTasks.join handle the errors instead

**Comment by artem-shelkovnikov** [comment]
Duplicate of https://github.com/elastic/enterprise-search-team/issues/5189
